course02/homework00/danielellis/vector.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <array>
+#include <exception>
+
+struct RanOutOfMemoryException : public std::exception
+{
+};
+
+class VectorInt
+{
+public:
+	size_t size() const
+	{
+		return mSize;
+	}
+
+	size_t capacity() const
+	{
+		return mMemory.max_size();
+	}
+
+	void push_back(const int& element)
+	{
+		if (size() >= mAllocatedMemory)
+		{
+			mAllocatedMemory *= 2;
+			if (mAllocatedMemory > capacity())
+				throw RanOutOfMemoryException();
+		}
+		mMemory.at(size()) = element;
+		mSize += 1;
+	}
+
+	int& back()
+	{
+		if (size() > 0)
+			return mMemory.at(size() - 1);
+		else
+			throw std::out_of_range("Our vector is empty so there is no last value");
```

Comment:
I would remove a level of indentation, but tats mostly style. I would also check first for errors, then the normal/happy case.